src/data/list/of_fn.lean

```diff
@@ -76,6 +77,15 @@ begin
   simp only [d_array.rev_iterate_aux, of_fn_aux, IH]
 end
 
+@[congr]
+theorem of_fn_congr {m n : ℕ} (h : m = n) (f : fin m → α) :
+  of_fn f = of_fn (λ i : fin n, f (fin.cast h.symm i)) :=
+begin
+  subst h,
+  simp_rw fin.cast_refl,
+  refl,
+end
```

Comment:
If you want to save having to mention a lemma by name:
```suggestion
by { cases h, congr, ext, simp }
```

Reply:
I think I'd rather have a squeezed simp to make it clear what's going on